src/gles_fns.rs

```diff
@@ -147,47 +147,102 @@ impl Gl for GlesFns {
 
     #[allow(unused_variables)]
     fn gen_queries(&self, n: GLsizei) -> Vec<GLuint> {
-        panic!("not supported")
+        if !self.ffi_gl_.GenQueriesEXT.is_loaded() {
+            return Vec::new();
+        }
+        unsafe {
+            let mut result = vec![0; n as usize];
+            self.ffi_gl_.GenQueriesEXT(n, result.as_mut_ptr());
+            return result;
+        }
     }
 
     #[allow(unused_variables)]
     fn begin_query(&self, target: GLenum, id: GLuint) {
-        panic!("not supported")
+        if !self.ffi_gl_.BeginQueryEXT.is_loaded() {
+            return;
+        }
+        unsafe {
+            self.ffi_gl_.BeginQueryEXT(target, id);
+        }
     }
 
     #[allow(unused_variables)]
     fn end_query(&self, target: GLenum) {
-        panic!("not supported")
+        if !self.ffi_gl_.EndQueryEXT.is_loaded() {
+            return;
+        }
+        unsafe {
+            self.ffi_gl_.EndQueryEXT(target);
+        }
     }
 
     #[allow(unused_variables)]
     fn query_counter(&self, id: GLuint, target: GLenum) {
-        panic!("not supported")
+        if !self.ffi_gl_.QueryCounterEXT.is_loaded() {
+            return;
+        }
+        unsafe {
+            self.ffi_gl_.QueryCounterEXT(id, target);
+        }
     }
 
     #[allow(unused_variables)]
     fn get_query_object_iv(&self, id: GLuint, pname: GLenum) -> i32 {
-        panic!("not supported")
+        if !self.ffi_gl_.GetQueryObjectivEXT.is_loaded() {
+            return 0;
+        }
+        unsafe {
+            let mut result = 0;
+            self.ffi_gl_.GetQueryObjectivEXT(id, pname, &mut result);
+            result
+        }
     }
 
     #[allow(unused_variables)]
```

Comment:
let's remove all of the `unused_variables` of the affected functions now

Reply:
Oh, I forgot to remove it. I will remove them.